core/src/main/java/io/questdb/cairo/wal/WalTxnDetails.java

```diff
@@ -176,93 +460,418 @@ private static WalEventCursor openWalEFile(Path tempPath, WalEventReader eventRe
+                final int key = entry.getKey() - cleanSymbolCount;
+                assert key == symbolIndex;
+                symbolIndex++;
+                entry.appendSymbolTo(symbolMem);
+            }
+
+            symbolIndexes.add(Numbers.decodeLowInt(symbolValueOffset));
+            symbolIndexes.add(Numbers.decodeHighInt(symbolValueOffset));
+
+            int count = symbolIndex;
+            // Update number of symbol column entries
+            symbolIndexes.set(entryBegins, count);
+            symbolCount++;
+            totalSymbolsSaved += count;
+        }
+
+        // Empty record, return -1, save space, don't serialize empty records.
+        if ((symbolCount == 0 || totalSymbolsSaved == 0) && !hasNull) {
+            symbolIndexes.setPos(startOffset);
+            return -1 - (currentSymbolIndexesStartOffset + startOffset);
         }
+
+        // Set the record length
+        symbolIndexes.set(startOffset, (int) symbolIndexes.size() - startOffset);
+        // Set the count of symbols
+        symbolIndexes.set(startOffset + 1, symbolCount);
+
+        return currentSymbolIndexesStartOffset + startOffset;
+    }
+
+    private void shiftSymbolStringsDataLeft(long shiftBytes) {
+        final long size = symbolStringsMem.getAppendOffset();
+        assert size >= shiftBytes;
+        if (size > shiftBytes) {
+            long address = symbolStringsMem.getPageAddress(0);
+            Vect.memcpy(address, address + shiftBytes, size - shiftBytes);
+            symbolStringsMem.jumpTo(size - shiftBytes);
+        } else {
+            symbolStringsMem.truncate();
+        }
+    }
+
+    private WalTxnDetailsSlice sortSliceByWalAndSegment(long startSeqTxn, int blockTransactionCount) {
+        assert blockTransactionCount > 1;
+        return txnSlice.of(startSeqTxn, blockTransactionCount);
     }
 
-    private void loadTransactionDetailsV2(TransactionLogCursor transactionLogCursor, long maxCommittedTimestamp) {
-        long runningMaxTimestamp = maxCommittedTimestamp;
-        int i = 0;
-        while (i++ < maxLookahead && transactionLogCursor.hasNext()) {
+    private class SymbolMapDiffCursorImpl implements SymbolMapDiffCursor {
+        private final SymbolMapDiffColumnRecord symbolMapDiff = new SymbolMapDiffColumnRecord();
+        int nextRecordLen;
+        private long hi;
+        private int lo;
+        private int symbolIndex;
+        private int symbolsCount;
+
+        @Override
+        public SymbolMapDiff nextSymbolMapDiff() {
+            if (symbolIndex++ < symbolsCount) {
+                assert lo <= hi;
+                symbolMapDiff.switchToColumnRecord(lo);
+                lo += 6;
+                return symbolMapDiff;
+            }
+            return null;
+        }
+
+        public void of(int startOffset) {
+            if (startOffset > -1) {
+                this.lo = startOffset;
+                int recordSize = WalTxnDetails.this.symbolIndexes.get(startOffset);
```

Comment:
confusing variable naming, usually "lo", "hi" and "size" are used in context of memory addresses, however here these are indexes, recordSize us actually a count? no?

Reply:
I renamed the variables, all code here measures in integers indexes.